app/src/main/java/sistemas/puc/com/finantialapp/IndiceFragment.java

```diff
@@ -1,37 +1,90 @@
 package sistemas.puc.com.finantialapp;
 
+import android.database.Cursor;
+import android.net.Uri;
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
+import android.support.v4.app.LoaderManager;
+import android.support.v4.content.CursorLoader;
+import android.support.v4.content.Loader;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ListView;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-
-import sistemas.puc.com.finantialapp.adapters.IndiceAdapter;
-import sistemas.puc.com.finantialapp.entities.IndiceItem;
+import sistemas.puc.com.finantialapp.adapters.AbstractCursorAdapter;
+import sistemas.puc.com.finantialapp.adapters.IndiceCursorAdapter;
+import sistemas.puc.com.finantialapp.data.FinantialContract.IndiceEntry;
 import sistemas.puc.com.finantialapp.model.Database;
+import sistemas.puc.com.finantialapp.util.DividerItemDecoration;
+
+public class IndiceFragment extends Fragment implements LoaderManager.LoaderCallbacks<Cursor> {
+
+    private static final Uri INDICE_URI = IndiceEntry.CONTENT_URI;
 
-public class IndiceFragment extends Fragment {
+    private static final String[] INDICE_COLUMNS = new String[]{
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry._ID,
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry.COLUMN_INDICE_CODE,
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry.COLUMN_INDICE_NAME,
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry.COLUMN_INDICE_DATE,
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry.COLUMN_INDICE_MONTH_RATE,
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry.COLUMN_INDICE_YEAR_RATE,
+            IndiceEntry.TABLE_NAME + "." + IndiceEntry.COLUMN_INDICE_TYPE,
+    };
 
-    IndiceAdapter m_indiceAdapter;
+    private RecyclerView m_recyclerView;
+    private AbstractCursorAdapter m_adapter;
+    private RecyclerView.LayoutManager m_layoutManager;
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
 
-        List<IndiceItem> indiceList = Database.getIndiceList();
+        View rootView = inflater.inflate(R.layout.fragment_indice, container, false);
 
-        m_indiceAdapter = new IndiceAdapter(getActivity(), indiceList);
+        // Update database on startup
+        Database.update(getContext());
 
-        View rootView = inflater.inflate(R.layout.fragment_indice, container, false);
+        DividerItemDecoration itemDecoration = new DividerItemDecoration(getActivity());
+        itemDecoration.setLastItemIncluded(false);
+        itemDecoration.setPaddingLeftDp(72);
+        itemDecoration.setPaddingRightDp(16);
```

Comment:
Maybe fix this hardcoded style in some way.
Add a todo comment if you're going to fix it later.